src/main/java/io/github/oliviercailloux/y2018/jconfs/ConferenceReader.java

```diff
@@ -84,16 +83,50 @@ public static Conference createConference(Reader read)
 		// add the others attributes
 		conf.setTitle(confCompo.getProperty("SUMMARY").getValue());
 		conf.setCountry(confCompo.getProperty("X-COUNTRY").getValue());
+		//convert DateTime pattern to LocalDate pattern
+
+		String stringDTSTART=confCompo.getProperty("X-DTSTART").getValue();
+		String stringDTEND=confCompo.getProperty("X-DTSTART").getValue();
+		DateTimeFormatter formatBefore=DateTimeFormatter.ofPattern("yyyyMMdd");
+		DateTimeFormatter formatAfter=DateTimeFormatter.ofPattern("dd/MM/yyy");
 		
+		LocalDate dtStart=LocalDate.parse(stringDTSTART,formatBefore);
+		LocalDate dtEnd=LocalDate.parse(stringDTEND,formatBefore);
 		
+		conf.setStartDate(dtStart.format(formatAfter));
+		conf.setEndDate(dtEnd.format(formatAfter));
 		
-		//convert DateTime pattern to LocalDate pattern
+		conf.setFeeRegistration(Double.parseDouble(confCompo.getProperty("X-FEE").getValue()));
+		conf.setCity(confCompo.getProperty("X-CITY").getValue());
+	
+		return conf;
+
+	}
+	
+	/**
+	 * We parse a calendar component to create a conference
+	 * @param confCompo it's a calendar component that contains the data of one conference
+	 * @return a conference
+	 * @throws IOException
+	 * @throws ParserException
+	 * @throws ParseException
+	 * @throws NumberFormatException
+	 */
+	public static Conference createConference(Component confCompo)
+			throws IOException, ParserException, ParseException, NumberFormatException {
+		Conference conf = null;
+		// the url is the primary key of a conference
+		URL confURL = new URL(confCompo.getProperty("URL").getValue());
+		conf = new Conference(confURL);
 
+		// add the others attributes
+		conf.setTitle(confCompo.getProperty("SUMMARY").getValue());
+		conf.setCountry(confCompo.getProperty("X-COUNTRY").getValue());
+		
+		//convert DateTime pattern to LocalDate pattern
 		String stringDTSTART=confCompo.getProperty("X-DTSTART").getValue();
 		String stringDTEND=confCompo.getProperty("X-DTSTART").getValue();
 
```

Comment:
Factoriser le code de conversion de date en en faisant une méthode appelée plusieurs fois, ne pas dupliquer le code.

Reply:
Il s'agit d'une erreur de saisie le second champs correspond à la date de fin noté "X-DTEND" , cela  a été modifié.

Reply:
Il faut quand-même une méthode qui s’occupe de la conversion de format de date.